sphinx/environment/adapters/toctree.py

```diff
@@ -315,19 +316,23 @@ def _toctree_entry(
             # empty toc means: no titles will show up in the toctree
             logger.warning(__('toctree contains reference to document %r that '
                               "doesn't have a title: no link will be generated"),
-                           ref, location=toctreenode)
-    except KeyError:
-        # this is raised if the included file does not exist
+                           ref, location=toctreenode, type='toc', subtype='no_title')
+    except KeyError as err:
+        # It may happen only after `directives/other.py` parsing when
+        # the included document changed/disappeared. Thus, it is a not correctable warning.
         ref_path = env.doc2path(ref, False)
         if excluded(ref_path):
             message = __('toctree contains reference to excluded document %r')
+            subtype = 'excluded'
```

Comment:
I see you are using this subtype but where is the toc.excluded actually used since it wasn't introduced in this PR?

Reply:
As the comment(L321) indicates, [/sphinx/directives/other.py#L141](/sphinx-doc/sphinx/blame/v7.2.5/sphinx/directives/other.py#L141)

Reply:
Ah sorry! actually I am reviewing everything on phone so it's quite hard to switch within the code (and then I forgot what I was looking at).